From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 10 Jul 2024 19:25:46 -0500
Subject: [PATCH] Add hack for steam users to skip hwcaps directories

This is an absolute, bona fide hack of the highest grade.

See https://github.com/ValveSoftware/steam-for-linux/issues/10556

---
 elf/ldconfig.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/elf/ldconfig.c b/elf/ldconfig.c
index b64c54b53e..fc01aa908f 100644
--- a/elf/ldconfig.c
+++ b/elf/ldconfig.c
@@ -304,6 +304,10 @@ add_glibc_hwcaps_subdirectories (struct dir_entry *entry, const char *path)
   /* glibc-hwcaps subdirectories do not nest.  */
   assert (entry->hwcaps == NULL);
 
+  char *steam_hack = getenv ("STEAM_HACK_IGNORE_HWCAPS");
+  if (steam_hack)
+    return;
+
   char *glibc_hwcaps;
   if (asprintf (&glibc_hwcaps, "%s/" GLIBC_HWCAPS_SUBDIRECTORY, path) < 0)
     error (EXIT_FAILURE, errno, _("Could not form glibc-hwcaps path"));
